pkg/cloudprovider/providers/azure/azure_storage.go

```diff
@@ -19,23 +19,53 @@ package azure
 import (
 	"fmt"
 
+	"github.com/Azure/azure-sdk-for-go/arm/storage"
+	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/golang/glog"
 )
 
+const (
+	defaultStorageAccountType  = string(storage.StandardLRS)
+	fileShareAccountNamePrefix = "f"
+)
+
 // CreateFileShare creates a file share, using a matching storage account
 func (az *Cloud) CreateFileShare(name, storageAccount, storageType, location string, requestGB int) (string, string, error) {
 	var err error
 	accounts := []accountWithLocation{}
-	if len(storageAccount) > 0 {
-		accounts = append(accounts, accountWithLocation{Name: storageAccount})
-	} else {
-		// find a storage account
-		accounts, err = az.getStorageAccounts()
+	if len(storageAccount) == 0 {
+		if location == "" {
+			location = az.controllerCommon.location
+		}
+		if storageType == "" {
+			storageType = defaultStorageAccountType
+		}
```

Comment:
This would be a behavior change for user specifying empty storageAccount and empty storageType:
Previously when any storage account is found, it will be used regardless of its type.
After this, only storage account with 'defaultStorageAccountType' type would be searched.

Reply:
fixed